File: framework-cs/apps/UnityExample/Assets/Samples/Devian Foundation/0.1.0/GamePackage/Runtime/Ability/AbilityBattleSkill.cs
using System;
using System.Collections.Generic;
using Devian.Domain.Game;

namespace Devian
{
    public sealed class AbilityBattleSkill : AbilityBattleBase
    {
        SKILL mTable = null;

        public string SkillId => mTable?.skill_id ?? string.Empty;
        public string NameId => mTable?.name_id ?? string.Empty;
        public IReadOnlyList<string> AffectList => mTable != null ? mTable.affect_list : Array.Empty<string>();

        public void Init(SKILL table)
        {
            mTable = table;
        }

        public override AbilityBase Clone()
        {
            var c = new AbilityBattleSkill();
            c.mTable = mTable;
            c.CopyStatsFrom(this);
            return c;
        }
    }
}

File: framework-ts/module/devian-domain-game/features/ability/AbilityBattleSkill.ts
import { SKILL } from '../../Generated/Game.g';
import { AbilityBattleBase } from './AbilityBattleBase';

export class AbilityBattleSkill extends AbilityBattleBase {
    private mTable: SKILL | null = null;

    get skillId(): string { return this.mTable?.skill_id ?? ''; }
    get nameId(): string { return this.mTable?.name_id ?? ''; }
    get affectList(): readonly string[] { return this.mTable?.affect_list ?? []; }

    init(table: SKILL): void {
        this.mTable = table;
    }

    clone(): AbilityBattleSkill {
        const c = new AbilityBattleSkill();
        c.mTable = this.mTable;
        c.copyStatsFrom(this);
        return c;
    }
}
